stl/inc/algorithm

```diff
@@ -4388,34 +4388,8 @@ constexpr pair<_FwdIt, _FwdIt> _Minmax_element_unchecked(_FwdIt _First, _FwdIt _
     // find smallest and largest elements, using _Pred
     pair<_FwdIt, _FwdIt> _Found(_First, _First);
 
-    if (_First != _Last) {
-        while (++_First != _Last) { // process one or two elements
-            _FwdIt _Next = _First;
-            if (++_Next == _Last) { // process last element
-                if (_DEBUG_LT_PRED(_Pred, *_First, *_Found.first)) {
-                    _Found.first = _First;
-                } else if (!_DEBUG_LT_PRED(_Pred, *_First, *_Found.second)) {
-                    _Found.second = _First;
-                }
-            } else { // process next two elements
-                if (_DEBUG_LT_PRED(_Pred, *_Next, *_First)) { // test _Next for new smallest
-                    if (_DEBUG_LT_PRED(_Pred, *_Next, *_Found.first)) {
-                        _Found.first = _Next;
-                    }
-                    if (!_DEBUG_LT_PRED(_Pred, *_First, *_Found.second)) {
-                        _Found.second = _First;
-                    }
-                } else { // test _First for new smallest
-                    if (_DEBUG_LT_PRED(_Pred, *_First, *_Found.first)) {
-                        _Found.first = _First;
-                    }
-                    if (!_DEBUG_LT_PRED(_Pred, *_Next, *_Found.second)) {
-                        _Found.second = _Next;
-                    }
-                }
-                _First = _Next;
-            }
-        }
+    while (++_First != _Last) { // O(N-1) applications of _Pred, where N = std::distance(_First, _Last)

```

Comment:
This breaks for `_First == _Last` as you are precrementing `_First`

Reply:
Right.
Post increment indeed.

Reply:
if you have a fail case it would be great! 
thanks

Reply:
Maybe moot, but this does also not work as it would skip `_First` from comparisons.